sdl_android/src/main/java/com/smartdevicelink/api/FileManager.java

```diff
@@ -0,0 +1,428 @@
+import com.smartdevicelink.proxy.rpc.enums.Result;
+import com.smartdevicelink.proxy.rpc.listeners.OnMultipleRequestListener;
+import com.smartdevicelink.proxy.rpc.listeners.OnRPCResponseListener;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * <strong>FileManager</strong> <br>
+ *
+ * Note: This class must be accessed through the SdlManager. Do not instantiate it by itself. <br>
+ *
+ * The SDLFileManager uploads files and keeps track of all the uploaded files names during a session. <br>
+ *
+ * We need to add the following struct: SDLFile<br>
+ *
+ * It is broken down to these areas: <br>
+ *
+ * 1. Getters <br>
+ * 2. Deletion methods <br>
+ * 3. Uploading Files / Artwork
+ */
+public class FileManager extends BaseSubManager {
+
+	private static String TAG = "FileManager";
+	private List<String> remoteFiles, uploadedEphemeralFileNames;
+	private WeakReference<Context> context;
+
+	FileManager(ISdl internalInterface, Context context) {
+
+		// setup
+		super(internalInterface);
+		this.context = new WeakReference<>(context);
+
+		// prepare manager - don't set state to ready until we have list of files
+		retrieveRemoteFiles();
+		uploadedEphemeralFileNames = new ArrayList<>();
+	}
+
+	// GETTERS
+
+	/**
+	 * Returns a list of file names currently residing on core
+	 * @return List<String> of remote file names
+	 */
+	public List<String> getRemoteFileNames() {
+		if (getState() != BaseSubManager.READY){
+			// error and dont return list
+			throw new IllegalArgumentException("FileManager is not READY");
+		}
+		// return list (this is synchronous at this point)
+		return remoteFiles;
+	}
+
+	private void retrieveRemoteFiles(){
+		remoteFiles = new ArrayList<>();
+		// hold list in remoteFiles class var
+		ListFiles listFiles = new ListFiles();
+		listFiles.setOnRPCResponseListener(new OnRPCResponseListener() {
+			@Override
+			public void onResponse(int correlationId, RPCResponse response) {
+				if(response.getSuccess()){
+					if(((ListFilesResponse) response).getFilenames() != null){
+						remoteFiles.addAll(((ListFilesResponse) response).getFilenames());
+					}
+					// on callback set manager to ready state
+					transitionToState(BaseSubManager.READY);
+				}else{
+					// file list could not be received
+					transitionToState(BaseSubManager.ERROR);
+				}
+			}
+		});
+		internalInterface.sendRPCRequest(listFiles);
+	}
+
+	// DELETION
+
+	/**
+	 * Attempts to delete the desired file from core, calls listener with indication of success/failure
+	 * @param fileName name of file to be deleted
+	 * @param listener callback that is called on response from core
+	 */
+	public void deleteRemoteFileWithName(final String fileName, final CompletionListener listener){
+		if(fileName == null || listener == null){
```

Comment:
Does it actually matter if the listener is null? Can't the developer just request the file be deleted without caring if it was successful?